packages/cxl-ui/src/components/cxl-marketing-nav.js

```diff
@@ -1,634 +1,356 @@
-    };
+  _renderSearch(group) {
+    if (this.minimal || group.name !== 'primary') return '';
 
-    // eslint-disable-next-line no-param-reassign
-    contextMenuItems = [backBtn, ...contextMenuItems];
-
-    return contextMenuItems;
+    return html`
+      <vaadin-menu-bar-button
+        theme="tertiary cxl-marketing-nav"
+        class="search-button"
+        @click=${this.toogleSearchDialog}
+      >
+        <a>
+          <vaadin-icon icon="lumo:search"></vaadin-icon> Search
+          <vaadin-icon icon="lumo:dropdown"></vaadin-icon>
+        </a>
+      </vaadin-menu-bar-button>
+    `;
   }
 
-  /**
-   * Create `<vaadin-context-menu>` elements, with children nesting support.
-   * Support `{ component: a }`.
-   *
-   * @private
-   * @see https://github.com/vaadin/vaadin-context-menu/issues/254
-   */
-  _createContextMenuItems(contextMenuItems) {
-    contextMenuItems.forEach((item, i, self) => {
-      if (item.children) {
-        // eslint-disable-next-line no-param-reassign
-        item.children = this._addBackButtonItem(item.children);
-        this._createContextMenuItems(item.children);
-      }
-
-      const menuItem = document.createElement('vaadin-context-menu-item');
-
-      if (item.sectionheader) {
-        menuItem.classList.add('section-header');
-
-        const label = document.createTextNode(item.text);
-        menuItem.appendChild(label);
-
-        // eslint-disable-next-line no-param-reassign
-        self[i] = { component: menuItem };
-      } else if (item.component === 'a') {
-        const link = document.createElement('a');
-
-        if (item.href) {
-          link.href = item.href;
-          link.innerHTML = item.text;
-        }
-
-        menuItem.appendChild(link);
-        // eslint-disable-next-line no-param-reassign
-        self[i] = { component: menuItem };
-      }
-
-      if (item.description) {
-        const descriptionItem = document.createElement('div');
-
-        descriptionItem.classList.add('vaadin-context-menu-item--description');
-
-        // Set to hidden, to calculate currently opened menu width and use it for description.
-        descriptionItem.hidden = true;
-
-        render(html`${unsafeHTML(item.description)}`, descriptionItem);
-
-        menuItem.appendChild(descriptionItem);
-        menuItem.classList.add('has-description');
-      }
-
-      if (item.component === 'back') {
-        const backBtn = document.createElement('vaadin-button');
-
-        backBtn.classList.add('context-menu-item-back-button');
-        backBtn.innerHTML = '<vaadin-icon icon="lumo:angle-left"></vaadin-icon> Back';
-
-        menuItem.classList.add('back-button-menu-item');
-        menuItem.appendChild(backBtn);
-
-        menuItem.addEventListener('click', this._onBackBtnClick.bind(this));
-
-        // eslint-disable-next-line no-param-reassign
-        self[i] = { component: menuItem };
-      }
-    });
-
-    return contextMenuItems;
+  toogleSearchDialog() {
```

Comment:
Double "G" (like Snoop Dogg) not "O"

```suggestion
  toggleSearchDialog() {
```